src/main/java/net/minecraftforge/items/wrapper/ShulkerItemStackInvWrapper.java

```diff
@@ -0,0 +1,247 @@
+
+        int m;
+        if (!stackInSlot.isEmpty())
+        {
+            if (stackInSlot.getCount() >= Math.min(stackInSlot.getMaxStackSize(), getSlotLimit(slot)))
+                return stack;
+
+            if (!ItemHandlerHelper.canItemStacksStack(stack, stackInSlot))
+                return stack;
+
+            if (!this.isItemValid(slot, stack))
+                return stack;
+
+            m = Math.min(stack.getMaxStackSize(), getSlotLimit(slot)) - stackInSlot.getCount();
+
+            if (stack.getCount() <= m)
+            {
+                if (!simulate)
+                {
+                    ItemStack copy = stack.copy();
+                    copy.grow(stackInSlot.getCount());
+                    itemStacks.set(slot, copy);
+                    this.setItemList(itemStacks);
+                }
+
+                return ItemStack.EMPTY;
+            }
+            else
+            {
+                // copy the stack to not modify the original one
+                stack = stack.copy();
+                if (!simulate)
+                {
+                    ItemStack copy = stack.split(m);
+                    copy.grow(stackInSlot.getCount());
+                    itemStacks.set(slot, copy);
+                    this.setItemList(itemStacks);
+                    return stack;
+                }
+                else
+                {
+                    stack.shrink(m);
+                    return stack;
+                }
+            }
+        }
+        else
+        {
+            if (!this.isItemValid(slot, stack))
+                return stack;
+
+            m = Math.min(stack.getMaxStackSize(), getSlotLimit(slot));
+            if (m < stack.getCount())
+            {
+                // copy the stack to not modify the original one
+                stack = stack.copy();
+                if (!simulate)
+                {
+                    itemStacks.set(slot, stack.split(m));
+                    this.setItemList(itemStacks);
+                    return stack;
+                }
+                else
+                {
+                    stack.shrink(m);
+                    return stack;
+                }
+            }
+            else
+            {
+                if (!simulate)
+                {
+                    itemStacks.set(slot, stack);
+                    this.setItemList(itemStacks);
+                }
+                return ItemStack.EMPTY;
+            }
+        }
+    }
+
+    @Override
+    @NotNull
+    public ItemStack extractItem(int slot, int amount, boolean simulate)
+    {
+        NonNullList<ItemStack> itemStacks = getItemList();
+        if (amount == 0)
+            return ItemStack.EMPTY;
+
+        ItemStack stackInSlot = itemStacks.get(slot);
+
+        if (stackInSlot.isEmpty())
+            return ItemStack.EMPTY;
+
+        if (simulate)
+        {
+            if (stackInSlot.getCount() < amount)
+            {
+                return stackInSlot.copy();
```

Comment:
This stack was just deserialized from NBT, there is no need to copy it.

Reply:
(Fixed by [4076409](https://github.com/MinecraftForge/MinecraftForge/pull/8827/commits/40764098c55531042d61406596c758b5a659664e))